src/main/java/com/o19s/es/explore/ExplorerQuery.java

```diff
@@ -106,10 +112,16 @@ public Weight createWeight(IndexSearcher searcher, ScoreMode scoreMode, float bo
             for (Term term : terms) {
                 TermStates ctx = TermStates.build(searcher.getTopReaderContext(), term, scoreMode.needsScores());
                 if(ctx != null && ctx.docFreq() > 0){
-                    TermStatistics tStats = searcher.termStatistics(term, ctx.docFreq(), ctx.totalTermFreq());
-                    df_stats.add(tStats.docFreq());
-                    idf_stats.add(sim.idf(tStats.docFreq(), searcher.collectionStatistics(term.field()).docCount()));
-                    ttf_stats.add(tStats.totalTermFreq());
+                    if (shard) {
+                        df_stats.add(ctx.docFreq());
+                        idf_stats.add(sim.idf(ctx.docFreq(), searcher.getIndexReader().getDocCount(term.field())));
+                        ttf_stats.add(ctx.totalTermFreq());
+                    } else {
+                        TermStatistics tStats = searcher.termStatistics(term, ctx.docFreq(), ctx.totalTermFreq());
+                        df_stats.add(tStats.docFreq());
+                        idf_stats.add(sim.idf(tStats.docFreq(), searcher.collectionStatistics(term.field()).docCount()));
+                        ttf_stats.add(tStats.totalTermFreq());
+                    }
```

Comment:
Hey Dan,
I have a question. Why do you need if/else here? I think we just need to use TermStatistics and everything will work correctly afterwards.
I mean if we just use TermStatistics (no need if/else) and do a request like below
```
POST my_index/_search?search_type=dfs_query_then_fetch
{
   "query": {
       "match_explorer": {
           "type": "sum_classic_idf",
           "query": {
               "match": {
                   "title": "when breath becomes air"
               }
           }
       }
   }
}
```
Both `docCount` and `docFreq` are over index.

If do a request like below (default if no `search_type`, then `search_type=query_then_fetch`)
```
POST my_index/_search
{
   "query": {
       "match_explorer": {
           "type": "sum_classic_idf",
           "query": {
               "match": {
                   "title": "when breath becomes air"
               }
           }
       }
   }
}
```
Both `docCount` and `docFreq` are over shard

Reply:
You're right. Cleanup inbound.

I'm having trouble getting the behavior to work correctly in the term stat query, I'm slightly unfamiliar with the `dfs_query_then_fetch` lifecycle so maybe something is incompatible.

Reply:
I don't know much about the code of term stat query. Hopefully, you will figure out something  with `dfs_query_then_fetch` for term stat query soon.